sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -383,6 +346,136 @@ protected EventProcessorClient()
             OwnsConnection = false;
         }
 
+        /// <summary>
+        ///   The function to be called just before event processing starts for a given partition.
+        /// </summary>
+        ///
+        /// <param name="context">The context in which the associated partition will be processed.</param>
+        ///
+        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        ///
+        protected override async Task InitializeProcessingForPartitionAsync(PartitionContext context)
+        {
+            try
+            {
+                var startingPosition = EventPosition.Earliest;
+
+                if (InitializeProcessingForPartitionAsyncHandler != null)
+                {
+                    var initializationContext = new InitializePartitionProcessingContext(context);
+                    await InitializeProcessingForPartitionAsyncHandler(initializationContext).ConfigureAwait(false);
+
+                    startingPosition = initializationContext.DefaultStartingPosition;
+                }
+
+                var ownership = InstanceOwnership[context.PartitionId];
+
+                if (ownership.Offset.HasValue)
+                {
+                    startingPosition = EventPosition.FromOffset(ownership.Offset.Value);
+                }
+                else if (ownership.SequenceNumber.HasValue)
+                {
+                    startingPosition = EventPosition.FromSequenceNumber(ownership.SequenceNumber.Value);
+                }
+
+                // TODO: it might be troublesome to let the users add running tasks by themselves.  If the user adds a custom
```

Comment:
It's a good question.   We can't prevent that, however.   The contract would be that if users choose to do something that isn't overriding our defined point for launching the task, they have to assume ownership of it.

When we loop back around to the base, can consider potentially exposing the `ActiveTasks` dictionary or some other means.

Reply:
We could let it like this for GA since it will be hidden, but we definitely need a way to allow the base class to signal when the task should stop. You suggested somewhere else we might want to make ActivePartitionProcessors a `ConcurrentDictionary<string, (Task, CancellationTokenSource)>`. I think it may be a solution.